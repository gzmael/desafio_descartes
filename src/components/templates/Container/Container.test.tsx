import { render, screen } from '@testing-library/react'

import { Container } from '.'

describe('Component/Template/Container', () => {
  const defaultClassName =
    'h-full w-full rounded-md bg-gradient-to-t from-[#BFDBEE] to-light-blue-500 p-[1px] shadow-lg'

  it('should render Container with children', () => {
    const props = {
      children: <p>Teste</p>,
    }

    const { container } = render(<Container {...props} />)

    expect(screen.getByTestId('container')).toBeInTheDocument()

    expect(container.firstChild).toHaveClass(defaultClassName)

    expect(container.firstChild).toMatchSnapshot()
  })
})
