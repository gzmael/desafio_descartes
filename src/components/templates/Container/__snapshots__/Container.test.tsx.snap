// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component/Template/Container > should render Container with children 1`] = `
<div
  class="h-full w-full rounded-md bg-gradient-to-t from-[#BFDBEE] to-light-blue-500 p-[1px] shadow-lg"
  data-testid="container"
>
  <div
    class="flex h-full w-full items-stretch justify-center overflow-hidden rounded-[4px] bg-gradient-to-b from-[#0CA7E5] to-blue-900"
  >
    <p>
      Teste
    </p>
  </div>
</div>
`;
